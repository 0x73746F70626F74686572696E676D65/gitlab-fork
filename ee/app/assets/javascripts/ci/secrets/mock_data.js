const mockSecretsData = [
  {
    name: 'runner-token-windows',
    labels: [],
    lastAccessed: '2024-03-19T20:55:08.551Z',
    createdAt: '2024-03-11T01:33:06.258Z',
  },
  {
    name: 'product/client-tokens',
    labels: ['env::staging', 'env::production'],
    lastAccessed: '2024-01-23T07:12:44.833Z',
    createdAt: '2024-01-22T08:04:26.024Z',
  },
  {
    name: 'product/server-tokens',
    labels: ['env::staging', 'env::production'],
    lastAccessed: '2024-01-22T07:12:44.833Z',
    createdAt: '2023-01-24T08:04:26.024Z',
  },
  {
    name: 'security-token-project2',
    labels: ['env::production', 'security'],
    lastAccessed: '2024-01-21T23:13:37.227Z',
    createdAt: '2024-01-19T11:59:43.516Z',
  },
  {
    name: 'personal-tokens',
    labels: [],
    lastAccessed: '2024-01-19T21:01:38.925Z',
    createdAt: '2024-01-17T05:51:54.602Z',
  },
  {
    name: 'token-deployment',
    labels: [],
    lastAccessed: '2024-01-16T23:13:31.548Z',
    createdAt: '2024-01-15T17:12:33.882Z',
  },
  {
    name: 'runner-token-mac',
    labels: [],
    lastAccessed: '2024-01-06T07:50:58.308Z',
    createdAt: '2023-01-15T16:35:38.326Z',
  },
  {
    name: 'aws-db-prod-credentials',
    labels: ['AWS'],
    lastAccessed: '2024-01-05T11:47:59.454Z',
    createdAt: '2024-01-04T10:37:41.135Z',
  },
  {
    name: 'runner-token-linux',
    labels: [],
    lastAccessed: '2023-12-30T16:41:08.551Z',
    createdAt: '2023-01-14T18:52:56.258Z',
  },
  {
    name: 'SSH-KEY-1',
    labels: [],
    lastAccessed: '2023-11-29T22:18:31.562Z',
    createdAt: '2023-11-30T01:29:14.394Z',
  },
  {
    name: 'group-ID',
    labels: ['shell'],
    lastAccessed: '2023-11-29T04:30:10.220Z',
    createdAt: '2023-01-21T17:34:33.544Z',
  },
  {
    name: 'security-token-project1',
    labels: ['security'],
    lastAccessed: '2023-11-28T04:30:10.220Z',
    createdAt: '2023-01-21T17:34:33.544Z',
  },
];

export const mockGroupSecretsData = mockSecretsData.map((secret, index) => ({
  key: `group_secret_${index}`,
  ...secret,
}));

export const mockProjectSecretsData = mockSecretsData.map((secret, index) => ({
  key: `project_secret_${index}`,
  ...secret,
}));
