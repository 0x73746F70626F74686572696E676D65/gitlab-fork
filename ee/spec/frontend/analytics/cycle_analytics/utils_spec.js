import {
  isStartEvent,
  isLabelEvent,
  getAllowedEndEvents,
  eventToOption,
  eventsByIdentifier,
  getLabelEventsIdentifiers,
} from 'ee/analytics/cycle_analytics/utils';
import { rawEvents } from './mock_data';

const startEvent = rawEvents[0];
const endEvent = rawEvents[1];
const labelEvent = rawEvents[5];
const labelEvents = [rawEvents[4], rawEvents[5]].map(i => i.identifier);

describe('Cycle analytics utils', () => {
  describe('isStartEvent', () => {
    it('will return true for a valid start event', () => {
      expect(isStartEvent(startEvent)).toEqual(true);
    });

    it('will return false for input that is not a start event', () => {
      [endEvent, {}, [], null, undefined].forEach(ev => {
        expect(isStartEvent(ev)).toEqual(false);
      });
    });
  });

  describe('isLabelEvent', () => {
    it('will return true if the given event identifier is in the labelEvents array', () => {
      expect(isLabelEvent(labelEvents, labelEvent.identifier)).toEqual(true);
    });
    it('will return false if the given event identifier is not in the labelEvents array', () => {
      [startEvent.identifier, null, undefined, ''].forEach(ev => {
        expect(isLabelEvent(labelEvents, ev)).toEqual(false);
      });
      expect(isLabelEvent(labelEvents)).toEqual(false);
    });
  });

  describe('eventToOption', () => {
    it('will return null if no valid object is passed in', () => {
      [{}, [], null, undefined].forEach(i => {
        expect(eventToOption(i)).toEqual(null);
      });
    });

    it('will set the "value" property to the events identifier', () => {
      rawEvents.forEach(ev => {
        const res = eventToOption(ev);
        expect(res.value).toEqual(ev.identifier);
      });
    });

    it('will set the "text" property to the events name', () => {
      rawEvents.forEach(ev => {
        const res = eventToOption(ev);
        expect(res.text).toEqual(ev.name);
      });
    });
  });

  describe('getLabelEventsIdentifiers', () => {
    it('will return an array of identifiers for the label events', () => {
      const res = getLabelEventsIdentifiers(rawEvents);
      expect(res.length).toEqual(labelEvents.length);
      expect(res).toEqual(labelEvents);
    });
    it('will return an empty array when there are no matches', () => {
      const ev = [{ _type: 'simple' }, { type: 'simple' }, { t: 'simple' }];
      expect(getLabelEventsIdentifiers(ev)).toEqual([]);
      expect(getLabelEventsIdentifiers([])).toEqual([]);
    });
  });

  describe('getAllowedEndEvents', () => {
    it('will return the relevant end events for a given start event identifier', () => {
      const se = rawEvents[4].allowedEndEvents;
      expect(getAllowedEndEvents(rawEvents, 'issue_label_added')).toEqual(se);
    });

    it('will return an empty array if there are no end events available', () => {
      ['cool_issue_label_added', [], {}, null, undefined].forEach(ev => {
        expect(getAllowedEndEvents(rawEvents, ev)).toEqual([]);
      });
    });
  });

  describe('eventsByIdentifier', () => {
    it('will return the events with an identifier in the provided array', () => {
      expect(eventsByIdentifier(rawEvents, labelEvents)).toEqual([rawEvents[4], rawEvents[5]]);
    });

    it('will return an empty array if there are no matching events', () => {
      [['lol', 'bad'], [], {}, null, undefined].forEach(items => {
        expect(eventsByIdentifier(rawEvents, items)).toEqual([]);
      });
      expect(eventsByIdentifier([], labelEvents)).toEqual([]);
    });
  });
});
