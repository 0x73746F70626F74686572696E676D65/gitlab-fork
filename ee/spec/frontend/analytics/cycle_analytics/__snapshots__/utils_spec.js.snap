// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value Stream Analytics utils transformRawStages retains all the stage properties 1`] = `
Array [
  Object {
    "custom": true,
    "description": "Time before an issue gets scheduled",
    "endEventIdentifier": "issue_first_added_to_board",
    "hidden": false,
    "id": 18,
    "legend": "Cool legend",
    "name": "Coolest beans stage",
    "slug": 18,
    "startEventIdentifier": "issue_first_mentioned_in_commit",
    "title": "Coolest beans stage",
  },
]
`;
